### PR TITLE
chore: review of plotly code

### PR DESCRIPTION
**Pull request by nickhenden**
This PR is to re-review the plotly code

### Related Issues

- Fixes #
- Closes #
- Relates to #

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [ ] ✨ New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🔧 Refactoring (no functional changes, no api changes)
- [ ] ⚡ Performance improvement
- [ ] 🧪 Test addition or improvement
- [ ] 🔨 Build/CI changes
- [ ] 🎨 Style/formatting changes

## Changes Made

- 
- 
- 

### Test Coverage
- [ ] Unit tests added/updated
- [ ] Integration tests added/updated
- [ ] Manual testing performed
- [ ] No tests needed (explain why below)

### Testing Details

**Test Cases:**
1. 
2. 
3. 

**Manual Testing Results:**

## Breaking Changes

- [ ] No breaking changes
- [ ] Breaking changes (describe below)

**Breaking Change Details:**

## Documentation
- [ ] Documentation updated
- [ ] README updated
- [ ] No documentation changes needed

**Documentation Changes:**

## Dependencies
- [ ] No new dependencies
- [ ] New dependencies added (list below)
- [ ] Dependencies updated (list below)
- [ ] Dependencies removed (list below)

**Dependency Changes:**

### Code Quality
- [ ] Code follows the project's [style guidelines](https://3.basecamp.com/4322319/buckets/25844562/documents/8777425747)
- [ ] Self-review of code completed
- [ ] Code is properly commented
- [ ] No unnecessary console.log/print statements
- [ ] No commented-out code (unless intentionally documented)

### Testing & Quality Assurance
- [ ] Tests pass locally
- [ ] Tests added for new functionality
- [ ] Edge cases considered and tested
- [ ] Error handling implemented where appropriate

### CODE REVIEW

**Review by rodriguesfred — COMMENTED** [review]
@nickhenden you made this in the wrong branch

File: src/afcharts/afcharts.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+            "plot_bgcolor": "rgba(0,0,0,0)",  # Makes plot area transparent
+            "margin": {  # Set margins around the plot area in pixels
+                "l": half_line,  # Left margin
+                "r": half_line,  # Right margin
+                "t": (base_size * 1.6) + (0.7 * (base_size * 1.6)) + half_line + base_size,
+                # Top margin is the size of the title + subtitle size + title_padding
+                # + space between title and subtitle (which is set by default)
+                "b": half_line,  # Bottom margin
+                "pad": 0,  # Padding between grid lines and the tick labels
+            },
```
Comment (wfahim) [inline src/afcharts/afcharts.py:152]: In R package for afcharts they have used half_line for t,b,l & r-`plot.margin = ggplot2::margin(half_line, half_line, half_line, half_line)`, but I can see you have used a different configuration for "Top" .I guess its something to relate with plotlys properties for the title text to sit inside the plotting container, not in the figure’s outer margin.

File: src/afcharts/afcharts.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+            "font": {
+                "color": "black",  # Text colour
+                "family": afcharts_font,  # Font
+                "size": base_size,
+            },  # Text size
+            "legend_title": None,  # Removes legend title
+            "legend": {
+                "borderwidth": 0,
+                "title": {"text": None},  # Removes legend title
+                "font": {"size": base_size * 1.2},  # Legend font size
```
[the remark this answers is not in the document]
Comment (rodriguesfred) [reply src/afcharts/afcharts.py:124]: Amended

File: src/afcharts/afcharts.py [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+                    "outlinewidth": 0,  # Width of the outline around the color bar
+                    "tickcolor": af_colour_values["chart_features"],  # Bar chart tick colour
+                    "ticklen": 6,  # Bar chart tick length
+                    "ticks": "outside",  # Bar chart tick position
+                }
+            },
+            "colorscale": {
+                "sequential": sequential,  # Sequential colour scale for low to high ranges
+                "sequentialminus": sequential_minus,  # Sequential palette for negatives
+                "diverging": diverging,  # Diverging colour scale
```
[the remark this answers is not in the document]
Comment (rodriguesfred) [reply src/afcharts/afcharts.py:112]: I'm going to not action this one as I know its incoming widder into the package so will wait for it to then refactor.

File: src/afcharts/afcharts.py [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+                "text": None,
+                "font": {
+                    "size": base_size * 1.6,
+                },  # Title font size and colour
+                "pad": {
+                    "t": half_line,
+                    "l": half_line,
+                    "r": half_line,
+                    "b": half_line,
+                },  # Padding above and below title
```
[the remark this answers is not in the document]
Comment (rodriguesfred) [reply src/afcharts/afcharts.py:166]: addressed in #45

File: src/afcharts/assets/pio_template.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+        "title": {
+            "text": None,
+            "font": {
+                "size": base_size * 1.6,
+            },  # Title font size and colour
+            "pad": {
+                "t": half_line,
+                "l": half_line,
+                "r": half_line,
+                "b": half_line,
```
[the remark this answers is not in the document]
Comment (rodriguesfred) [reply src/afcharts/assets/pio_template.py:96]: addressed in #45

File: src/afcharts/afcharts.py (hunk 1 above)
Comment (nickhenden) [reply src/afcharts/afcharts.py:152]: You can forget about the title, subtitle and source. For accessibility reasons, they should be included in the surrounding text rather than embedded in the plot. They recently changed the R cookbook to do the same thing.

In which case we don't need to modify the margins to allow for titles etc. Users can do that themselves if they really need to, but they shouldn't
